File: transfusion-center/src/main/java/isa/transfusioncenter/Model/Questionaire.java
package isa.transfusioncenter.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "questionaires")
@Entity
public class Questionaire {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private boolean answerToFirstQuestion;
    @Column
    private boolean answerToSecondQuestion;
    @Column
    private boolean answerToThirdQuestion;
    @Column
    private boolean answerToFourthQuestion;
    @Column
    private boolean answerToFifthQuestion;
    @Column
    private boolean answerToSixthQuestion;
    @Column
    private boolean answerToSeventhQuestion;
    @Column
    private boolean answerToEighthQuestion;
    @Column
    private boolean answerToNinethQuestion;
    @Column
    private boolean answerToTenthQuestion;
    @Column
    private boolean answerToEleventhQuestion;
    @Column
    private boolean answerToTwelfthQuestion;
    @Column
    private boolean answerToThirteenthQuestion;
    @Column
    private boolean answerToFourteenthQuestion;
    @Column
    private boolean answerToFifteenthQuestion;
    @Column
    private boolean answerToSixteenthQuestion;
    @Column
    private boolean answerToSeventeenthQuestion;
    @Column
    private boolean answerToEighteenthQuestion;
    @Column
    private boolean answerToNineteenthQuestion;
    @Column
    private boolean answerToTwentiethQuestion;
    @Column
    private boolean answerToTwentyFirstQuestion;
    @Column
    private boolean answerToTwentySecondQuestion;
    @Column
    private boolean answerToTwentyThirdQuestion;
    @Column
    private boolean answerToTwentyFourthQuestion;
    @Column
    private boolean answerToTwentyFifthQuestion;
    @Column
    private boolean answerToTwentySixthQuestion;

    @OneToOne(cascade = CascadeType.MERGE)
    private RegisteredUser user;
}

File: frontend-transfusion-center/src/app/models/questionaire.ts
import { RegisteredUser } from './registered-user';

export interface Questionaire {
  id: Number;
  answerToFirstQuestion: boolean;
  answerToSecondQuestion: boolean;
  answerToThirdQuestion: boolean;
  answerToFourthQuestion: boolean;
  answerToFifthQuestion: boolean;
  answerToSixthQuestion: boolean;
  answerToSeventhQuestion: boolean;
  answerToEighthQuestion: boolean;
  answerToNinethQuestion: boolean;
  answerToTenthQuestion: boolean;
  answerToEleventhQuestion: boolean;
  answerToTwelfthQuestion: boolean;
  answerToThirteenthQuestion: boolean;
  answerToFourteenthQuestion: boolean;
  answerToFifteenthQuestion: boolean;
  answerToSixteenthQuestion: boolean;
  answerToSeventeenthQuestion: boolean;
  answerToEighteenthQuestion: boolean;
  answerToNineteenthQuestion: boolean;
  answerToTwentiethQuestion: boolean;
  answerToTwentyFirstQuestion: boolean;
  answerToTwentySecondQuestion: boolean;
  answerToTwentyThirdQuestion: boolean;
  answerToTwentyFourthQuestion: boolean;
  answerToTwentyFifthQuestion: boolean;
  answerToTwentySixthQuestion: boolean;
  user: RegisteredUser;
}
